service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/workspace/FindResourcesToCloneStep.java

```diff
@@ -41,12 +45,18 @@ public StepResult doStep(FlightContext context) throws InterruptedException, Ret
       batch = resourceDao.enumerateResources(sourceWorkspaceId, null, null, offset, limit);
       offset += limit;
       final List<WsmResource> cloneableResources =
-          batch.stream().filter(FindResourcesToCloneStep::isCloneable).collect(Collectors.toList());
+          batch.stream().filter(FindResourcesToCloneStep::isCloneable).toList();
       cloneableResources.forEach(
           r -> result.add(new ResourceWithFlightId(r, context.getStairway().createFlightId())));
     } while (batch.size() == limit);
-    context.getWorkingMap().put(ControlledResourceKeys.RESOURCES_TO_CLONE, result);
 
+    // sort the resources by stewardship type reversed, so reference types go first
+    result.sort(
+        Comparator.comparing(
+            r -> r.getResource().getStewardshipType().toString(), Comparator.reverseOrder()));
```

Comment:
if you didn't want to rely on on the enum names / order not changing, perhaps you could make a comparator function that just returns 1 for reference, and -1 for anything else (or something similar to meet the contract)?

Reply:
does the order for other stewardship type besides referenced (if there are multiple matters)? we can also just order by enum values and add a comment that it shouldn't be swapped.

Reply:
If there were multiple, that might matter. I tend to avoid dealing with enum ordinals, though that's a valid way to do it too.

Note that we'll eventually maybe do the processing in parallel, so that even if we do referenced types first, we might start the controlled types before the last ref type is finished.